File: src/Problem020FactorialDigitSum.java
public static BigInteger factorial(int n) {
  BigInteger result = BigInteger.ONE;

  for (int i = 2; i <= n; i++) result = result.multiply(BigInteger.valueOf(i));

  return result;
}

void main() {
  String factorial = factorial(100).toString();
  int sum = 0;

  for (char c : factorial.toCharArray()) sum += c - '0';

  IO.println(factorial);
  IO.println(sum);
}

File: src/Problem020FactorialDigitSum.kt
import java.math.BigInteger

fun factorial(n: BigInteger): BigInteger {
  return if (n == BigInteger.ZERO) BigInteger.ONE
  else n * factorial(n - BigInteger.ONE)
}

fun main() {
  val factorial = factorial(BigInteger.valueOf(100))

  val sum = factorial.toString()
    .toList()
    .map { c -> c.digitToInt() }
    .reduce { x, y -> x + y }
  println(factorial)
  println(sum)
}
